### ISSUE TITLE
Provide type declarations for Node.js bindings

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/osrm-backend#4286]
We should provide type declarations for our Node.js bindings. We more or less already do this in our [api docs](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md) by spelling out the types in docs and linking to their mdn page. Once we have a declaration file clients can get their code checked at compile time when interacting with our Node.js bindings.

- https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html
- https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html

tl;dr

1/ add `lib/index.d.ts`

2/ add type declarations, e.g.

```typescript
interface RouteStep {
  distance: number;
  duration: number;
}
```

3/ optional: add comments / move our existing docs over here and let a tool check and generate api docs

**Comment by daniel-j-h** [comment]
Not actionable; no one cared for it.
